### PR TITLE
[slonik] Update types to v21.4

### PR DESCRIPTION
**Pull request by bram-l**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
https://github.com/gajus/slonik/compare/e7126935ec8642c4f01a9aead0498a16dfa99151...master#diff-3a699a0719518f967e9b2df3f5549e02
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/slonik/index.d.ts [hunk 1]
```diff
@@ -265,11 +267,18 @@ export interface QueryContextType {
...
     queryId: QueryIdType;
-    stackTrace: CallSiteType[] | null;
+
     /**
      * `process.hrtime.bigint()` for when query was received.
      */
     queryInputTime: number;
+
+    /** Object used by interceptors to assign interceptor-specific, query-specific context. */
+    // tslint:disable-next-line ban-types
```
Comment (mmkal) [inline types/slonik/index.d.ts]: Why not use `object`?

File: types/slonik/index.d.ts [hunk 2]
```diff
@@ -367,28 +376,37 @@ export interface InterceptorType {
...
+    beforeTransformQuery?: (
+        queryContext: QueryContextType,
+        query: QueryType
+    ) => MaybePromiseType<null>;
     queryExecutionError?: (
         queryContext: QueryContextType,
         query: QueryType,
         error: SlonikError
-    ) => MaybePromiseType<void>;
+    ) => MaybePromiseType<null>;
```
Comment (mmkal) [inline types/slonik/index.d.ts]: Is this a change from v21? Could you link to the docs?

**Review by mmkal — COMMENTED** [review]
Mostly looks good, but I haven't used v21 yet so maybe @sebald or @gajus could weigh in?

File: types/slonik/index.d.ts (hunk 2 above)
Comment (gajus) [reply types/slonik/index.d.ts]: https://github.com/gajus/slonik/blob/1ec5326171fa79135db93f92d35c5f8c628eb7fd/src/types.js#L468-L472

That is correct. It was one of the breaking changes.

File: types/slonik/index.d.ts (hunk 1 above)
Comment (gajus) [reply types/slonik/index.d.ts]: I am not familiar with TS enough to comment, but pretty sure it should be `any`.

It is basically an object that interceptors use to assign any arbitrary properties associated with the query.

Comment (bram-l) [reply types/slonik/index.d.ts]: I've updated the `sandbox` type with `Record<string, any>`, which provides slightly better type safety compared to `any`

**Review by mmkal — APPROVED** [review]

**Pull request merged**